flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/compiler/NFACompiler.java

```diff
@@ -150,6 +160,29 @@ long getWindowTime() {
 		}
 
 		/**
+		 * Check pattern after match skip strategy.
+		 */
```

Comment:
We only need to check the skip strategy before compile the `Pattern` to `NFA`, I think it's more reasonable to place it here. Also, we need to check whether the `patternName` field in the `AfterMatchSkipStrategy` is a valid reference, which can not be done easily in `Pattern` class.